beacon/sync/src/main/java/tech/pegasys/teku/beacon/sync/historical/HistoricalBlockSyncService.java

```diff
@@ -170,6 +181,10 @@ private void fetchBlocks() {
             () -> {
               if (isSyncDone()) {
                 stop().ifExceptionGetsHereRaiseABug();
+
+                reconstructHistoricalStatesService
+                    .doStart()
+                    .finish(error -> LOG.error(error.getMessage())); // fixme log error not working
```

Comment:
This should be `.start()` and we should only start the service if reconstructing historical states is enabled (it's safe to stop it regardless - the `Service` base class makes sure stop() does nothing if the service isn't already running.